Automattic/wp-calypso
# Force no site selection on `/theme` and `/themes`

After deploying https://github.com/Automattic/wp-calypso/pull/87000, I noticed that the left side navigation appeared different on production (showing a selected site) than it did on local/PR preview (showing "All sites"), as you can see on the screenshot:

![Screenshot from 2024-02-15 10-51-53](https://github.com/Automattic/wp-calypso/assets/8511199/7f5ec48f-0b79-4e9e-aa0d-e01ebc13f332)

## Proposed Changes

On prod, `CurrentSite` has a `selectedSite` object, whereas on local it's `null`, and the `selectedSite` prop comes from `state.ui.selectedSiteId`, so this PR attempts to fix that by setting the selected site to null with the `noSite` middleware.

## Testing Instructions

1. Using an account that has more than one site, go to the PR preview link, and then to `/themes?flags=-layout/dotcom-nav-redesign`
2. Check that you see "All sites" at the top of the left side navigation:

![image](https://github.com/Automattic/wp-calypso/assets/8511199/3e3b6c65-ac62-435c-b2c3-93e6df48a477)

## Pre-merge Checklist
